src/app/mainwindow.cpp

```diff
@@ -315,20 +315,19 @@ bool MainWindow::event(QEvent* event)
 
 void MainWindow::closeEvent(QCloseEvent* event)
 {
-    if (isVisible()) {
-        saveState();
-        d.save = false;
-
-        foreach (IrcConnection* connection, d.connections) {
-            connection->quit(qApp->property("description").toString());
-            connection->close();
-        }
+    saveState();
+    d.save = false;
 
-        // let the sessions close in the background
-        hide();
-        event->ignore();
-        QTimer::singleShot(1000, qApp, SLOT(quit()));
```

Comment:
yuk... this was a quick and dirty way to let open connections close gracefully in the background. :) not sure why it does not simply wait for the connections to be closed. libcommuni could probably provide some convenience functions for waiting certain states, similarly to QAbstractSocket::waitForConnected() and QAbstractSocket::waitForDisconnected(). what do you think?

Reply:
Yeah, some locking function with a timer or a signal sounds like a better solution than arbitrary timer.